guide/api-design-guidelines.md

```diff
@@ -0,0 +1,181 @@
+
+</details>
+
+## Introduction
+
+These are prescriptive guidelines that Radius contributors MUST follow while designing APIs to ensure that customers have a great experience. These guidelines help make Radius APIs:
+- Developer friendly via consistent patterns
+- Efficient & cost-effective
+
+Technology and software is constantly changing and evolving, and as such, this is intended to be a living document. [Open an issue](https://github.com/radius-project/design-notes/issues) to suggest a change or propose a new idea. 
+
+### Prescriptive Guidance
+This document offers prescriptive guidance labeled as follows:
+
+:white_check_mark: **DO** adopt this pattern. If you feel you need an exception, present your reason at a design review discussion **prior** to implementation.
+
+:ballot_box_with_check: **YOU SHOULD** adopt this pattern. If not following this advice, you MUST disclose your reason during a design review discussion.
+
+:heavy_check_mark: **YOU MAY** consider this pattern if appropriate to your situation.
+
+:warning: **YOU SHOULD NOT** adopt this pattern. If not following this advice, you MUST disclose your reason during a design review discussion.
+
+:no_entry: **DO NOT** adopt this pattern. If you feel you need an exception, present your reason at a design review discussion **prior** to implementation.
+
+## API Foundation:
+
+Radius provides an HTTP-based API that deploys and manages cloud-native applications as well as on-premise or cloud resources.
+The overall design of the Radius API is based on the Azure Resource Manager API (ARM). Radius generalizes the design of ARM, removes proprietary Azure concepts, and extends the ARM contract to support the conventions of other resource managers like Kubernetes or AWS Cloud-Control. 
+
+Radius utilizes TypeSpec for describing API model definitions and operations, providing a consistent approach to defining the structure of API payloads and interactions. The TypeSpec specification serves as the source to generate corresponding OpenAPI 2.0 (swagger) API documentation.
+
+Reference: 
+- [Radius API](https://docs.radapp.io/concepts/technical/api/)
+- [Radius Architecture](https://docs.radapp.io/concepts/technical/architecture/)
+
+## Guidelines
+
+### Resource Property Design
+
+<a href="#secrets" name="secrets"></a>
+### Secrets
+
+Radius enables users to securely store sensitive data, such as passwords, OAuth tokens, and SSH keys, in [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+The secrets are stored in secret store resource [Applications.Core/secretStores](https://docs.radapp.io/reference/resource-schema/core-schema/secretstore/)
+
+Our TypeSpec definitions provide the `SecretReference` and `SecretConfig` model types to standardize usage. These are prescribed to ensure sensitive information is handled securely and consistently across different components and resources in Radius. The following are their definitions along with usage examples:
```

Comment:
Should we add a link to the most recent **SecretReference** and **SecretConfig** in the code? Asking in case if there is an update and if we forget to update the models in this document below.